code/modules/clothing/head/_head.dm

```diff
@@ -27,4 +29,23 @@
 	..()
 	if(ismob(loc))
 		var/mob/M = loc
-		M.update_inv_head()
\ No newline at end of file
+		M.update_inv_head()
+
+/obj/item/clothing/head/throw_impact(atom/hit_atom, datum/thrownthing/thrownthing)
+	..()
+	if(thrownthing.target_zone && thrownthing.target_zone != BODY_ZONE_HEAD)
+		return
+	if(iscarbon(hit_atom))
+		var/mob/living/carbon/H = hit_atom
+		if(H.head)

```

Comment:
This is the appropriate typecheck spot, variable assignment should always type-check the object to the variables type.
```suggestion
		if(H.head && istype(H.head, /obj/item/clothing/head))
```

Reply:
https://github.com/tgstation/tgstation/pull/41005/files#discussion_r226858705

Reply:
It doesn't matter if before/after as long as it's not calling anything exclusive to the type.